feat(docs): major rewrite using vue-press

olivierlambert opened the pull request:
### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [ ] enhancement/bug fix entry added
  - [ ] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [ ] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

olivierlambert committed:
add minimal package and yarn.lock for VuePress

olivierlambert committed:
first content and reorg for vuepress

olivierlambert committed:
auto style? dunno

olivierlambert committed:
better doc content and org

olivierlambert committed:
add more content

olivierlambert committed:
improve content

olivierlambert committed:
continue to work on management part of the doc

olivierlambert committed:
add content

olivierlambert committed:
reorg and add content

olivierlambert committed:
better doc and reorg

olivierlambert committed:
section for support tunnel

olivierlambert committed:
add also xoa check section

olivierlambert committed:
add title and desc

olivierlambert commented on docs/.vuepress/styles/palette.styl:
```diff
@@ -0,0 +1 @@
+$accentColor = #cc584c
```
This is how we get the XCP-ng color for everything

olivierlambert commented on docs/backup.md:
```diff
@@ -0,0 +1,21 @@
+# Backup
```
This is the "index" of the whole backup section, giving an overview on the rest

olivierlambert commented on docs/backup_reports.md:
```diff
@@ -1,8 +1,8 @@
-# How to configure backup reports
+# Backup reports
```
In this file I mainly avoided title info duplication and fixed broken links

olivierlambert commented on docs/backups.md:
```diff
@@ -1,44 +1,80 @@
-# Backups
+# Concepts
```
This file became "concept" page explaining all common backup features (schedule, restore etc.)

julien-f commented on package.json:
```diff
@@ -64,6 +65,8 @@
     "clean": "scripts/run-script --parallel clean",
     "dev": "scripts/run-script --parallel dev",
     "dev-test": "jest --bail --watch \"^(?!.*\\.integ\\.spec\\.js$)\"",
+    "docs:dev": "vuepress dev docs",
+    "docs:build": "vuepress build docs",
```
Why not create `docs/package.json` directly?

```json
    "docs:dev": "vuepress dev .",
    "docs:build": "vuepress build .",
```

olivierlambert replied:
I'm following this: https://vuepress.vuejs.org/guide/getting-started.html#inside-an-existing-project

And the way it worked for XCP-ng doc. I'd like to avoid "taking risks" on using something else, except if you tell me exactly how to do it.

olivierlambert committed:
continue to improve doc

olivierlambert committed:
add reverse proxy

olivierlambert committed:
add tips in configuration

olivierlambert committed:
add proxy

olivierlambert committed:
doc is now complete

olivierlambert committed:
minor fix

olivierlambert committed:
add RAM enabled backup

olivierlambert committed:
add packages for Fedora/Centos install

olivierlambert committed:
forgot redis!

olivierlambert committed:
Add correct syntax for reverse proxy and other minor things

jcharaoui committed:
Specify recommended version of Node.js (#4967)

At this time, Node.js 10 can still be described as an LTS version,
according to https://nodejs.org/en/about/releases/

Update example to latest nodejs 12 minor release.

julien-f commented on docs/backup_troubleshooting.md:
```diff
@@ -1,16 +1,16 @@
 While a backup job is running, you should see activity in the "Tasks" view (Menu/Tasks), like this:
 
-![](assets/export_task.png)
+![](https://xen-orchestra.com/docs/assets/export_task.png)
```
Why use an absolute path here?

julien-f commented on docs/configuration.md:
```diff
@@ -105,6 +111,57 @@ You can add this at the end of your config file:
 
 On XOA, the log file for XO-server is in `/var/log/syslog`. It contains all the server information returned and can be a real help when you have trouble.
+
+## Reverse proxy
```
What about old files?

Example: `docs/reverse_proxy.md`.

olivierlambert replied:
We keep them for a bit for retro-compatibility issues (if we have old links around, it will continue to work)

julien-f replied:
I think it would be better to remove them and to handle this at the HTTP server level, by generating both the new and old docs, and use the old as a fallback.

olivierlambert replied:
Sounds like more than we can achieve on short term regarding all other priorities.

julien-f commented on docs/sdn_controller.md:
```diff
@@ -26,11 +24,14 @@ In the network creation view:
 - All hosts in a private network must be able to reach the other hosts' management interface.
   > The term ‘management interface’ is used to indicate the IP-enabled NIC that carries the management traffic.
 - Only 1 encrypted GRE network and 1 encrypted VxLAN network per pool can exist at a time due to Open vSwitch limitation.
+  :::
```
Note that this is non-standard syntax and it's formatting may be broken in the future by more standard tools we are using (eg Prettier).

olivierlambert replied:
Ok

julien-f commented on docs/sdn_controller.md:
```diff
@@ -1,17 +1,15 @@
+## Global Private Networks
 
-![](./assets/sdn-controller.png)
+![](https://xen-orchestra.com/blog/content/images/2019/08/gpn.png)
```
IMHO, it's a bad idea to depend on external resources, especially for trivial things like pictures that could be duplicated here.

olivierlambert replied:
Understood

julien-f commented on docs/updater.md:
```diff
@@ -1,10 +1,12 @@
+# Updates
 
-The updater is the central piece that keeps your XO Appliance running the latest Xen Orchestra code.
+All updates are pushed trhough the updater, which is the central piece that keeps your XO Appliance running the latest Xen Orchestra code.
```
```suggestion
All updates are pushed through the updater, which is the central piece that keeps your XO Appliance running the latest Xen Orchestra code.
```

julien-f commented on docs/updater.md:
```diff
@@ -40,29 +42,30 @@ However, if you want to start a manual check, you can do it by clicking on the "
 The stable channel is intended to be a version of Xen Orchestra that is already **one month old** (and therefore will benefit from one month of community feedback and various fixes). This way, users more concerned with the stability of their appliance will have the option to stay on a slightly older (and tested) version of XO (still supported by our pro support).
 
-##### Latest
+#### Latest ![](https://camo.githubusercontent.com/0ef9d8d2b62afd14829822d40cc5a949a7bcdedc/68747470733a2f2f62616467656e2e6e65742f62616467652f6368616e6e656c2f6c61746573742f79656c6c6f77)
```
```suggestion
#### Latest ![](https://badgen.net/badge/channel/latest/yellow)
```

julien-f commented on docs/updater.md:
```diff
@@ -40,29 +42,30 @@ However, if you want to start a manual check, you can do it by clicking on the "
 In Xen Orchestra, you can make a choice between two different release channels.
 
-##### Stable
+#### Stable ![](https://camo.githubusercontent.com/3eeb45846aece7f2db1348c8aa38191d18d66aec/68747470733a2f2f62616467656e2e6e65742f62616467652f6368616e6e656c2f737461626c652f677265656e)
```
```suggestion
#### Stable ![](https://badgen.net/badge/channel/stable/green)
```

olivierlambert committed:
added ALT feature

olivierlambert committed:
add SDN warning for MTU

olivierlambert committed:
PR review

julien-f committed:
Merge branch 'master' into olt-new-docs
